target-app/Home/Views/HomeViewController.swift

```diff
@@ -24,22 +45,38 @@ class HomeViewController: UIViewController {
         setupMapConstraints()
     }
     
+    override func viewDidAppear(_ animated: Bool) {
+        LocationManager.shared.requestLocationAuthorization()
+        
+        if CLLocationManager.locationServicesEnabled() {
```

Comment:
this is a bit confusing, you are using a LocationManager class for authorization and CLLocationManager for location related stuff. This means you will need to use both classes, and multiple instances of CLLocationManager in different places.
It is probably a good idea to have a private instance of CLLocationManager in the LocationManager class and use only that one.